core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -66,24 +67,19 @@ oppia.directive('topNavigationBar', [
           $scope.onLogoutButtonClicked = function() {
             $window.localStorage.removeItem('last_uploaded_audio_lang');
           };
-
-          $scope.profileDropdownIsActive = false;
-          $scope.onMouseoverProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
-            $scope.profileDropdownIsActive = true;
-          };
-          $scope.onMouseoutProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
-            $scope.profileDropdownIsActive = false;
-          };
-          $scope.onMouseoutDropdownMenuAbout = function(evt) {
-            angular.element(evt.currentTarget)[0].blur();
+          $scope.openSubMenu = function (menu) {
+            angular.element('.nav a').blur();
+            $scope.activeMenu = menu;
           };
-          $scope.onMouseoverDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
-          };
-          $scope.onMouseoutDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
+          $scope.closeSubMenu = function (evt) {
+            $scope.activeMenu = '';
+            if (evt.currentTarget.nodeName === 'UL') {
+              if (angular.element(evt.currentTarget).closest('li')
```

Comment:
Just FYI, I think we have a "focusService" in app.js that might provide a cleaner way to handle focus stuff. Not sure if it's useful (and don't worry about using it if it's not), just wanted to make you aware of it as something to investigate. (If you can use it, that's great, since my concern here is that if the DOM changes in the future and the JS doesn't, things are gonna break.)

Reply:
Thanks for the pointer.I will have a look at it.